ompi/mca/coll/han/coll_han_alltoallv.c

```diff
@@ -509,6 +549,15 @@ static int decide_to_use_smsc_alg(
     our execution time, which is <1/10 of the "basic" algorithm.
     */
 
+   if (sbuf == MPI_IN_PLACE) {
```

Comment:
Are you sure?  I was reading this passage of the 4.1 report about the alltoallv and alltoall collectives:

> The “in place” option for intra-communicators is specified by passing MPI_IN_PLACE to the argument sendbuf at *all* MPI processes.

Page 221 line 32.

and line 38:
> Advice to users. Specifying the “in place” option (which must be given on all MPI processes)...

Reply:
As a side note @wenduwan has reasonably pointed out we should probably cache the status of MCA_SMSC_FEATURE_CAN_MAP at comm creation so we can also make sure that all nodes have the same set of modules that get loaded.  I feel like for non-heterogeneous jobs, this shouldn't be required, but perhaps it is good defensive practice otherwise some of the ranks go execute an entirely different algorithm.

Reply:
I stand corrected, for many-to-many collectives `MPI_IN_PLACE` must be indeed provided to **all** processes.